tests/pipeline/test_statistical.py

```diff
@@ -0,0 +1,759 @@
+"""
+Tests for statistical pipeline terms.
+"""
+from numpy import (
+    arange,
+    full,
+    full_like,
+    nan,
+    where,
+)
+from pandas import (
+    DataFrame,
+    date_range,
+    Int64Index,
+    Timestamp,
+)
+from pandas.util.testing import assert_frame_equal
+from scipy.stats import linregress, pearsonr, spearmanr
+
+from zipline.assets import Equity
+from zipline.errors import IncompatibleTerms, NonExistentAssetInTimeFrame
+from zipline.pipeline import CustomFactor, Pipeline
+from zipline.pipeline.data import USEquityPricing
+from zipline.pipeline.data.testing import TestingDataSet
+from zipline.pipeline.engine import SimplePipelineEngine
+from zipline.pipeline.factors import (
+    Returns,
+    RollingLinearRegressionOfReturns,
+    RollingPearsonOfReturns,
+    RollingSpearmanOfReturns,
+)
+from zipline.pipeline.loaders.frame import DataFrameLoader
+from zipline.pipeline.sentinels import NotSpecified
+from zipline.testing import (
+    AssetID,
+    AssetIDPlusDay,
+    check_arrays,
+    make_alternating_boolean_array,
+    make_cascading_boolean_array,
+    parameter_space,
+)
+from zipline.testing.fixtures import (
+    WithSeededRandomPipelineEngine,
+    WithTradingEnvironment,
+    ZiplineTestCase,
+)
+from zipline.utils.numpy_utils import (
+    bool_dtype,
+    datetime64ns_dtype,
+    float64_dtype,
+)
+
+
+class StatisticalBuiltInsTestCase(WithTradingEnvironment, ZiplineTestCase):
```

Comment:
This whole test suite has just been moved from `test_engine`, there is no logic change here.